libdevice/nativecpu_utils.cpp

```diff
@@ -77,9 +77,9 @@ __spirv_ControlBarrier(uint32_t Execution, uint32_t Memory,
 
 namespace ncpu_types {
 template <class T> struct vtypes {
-  using v2 = typename sycl::detail::VecStorage<T, 2>::DataType;
-  using v4 = typename sycl::detail::VecStorage<T, 4>::DataType;
-  using v8 = typename sycl::detail::VecStorage<T, 8>::DataType;
+  using v2 = typename sycl::vec<T, 2>::vector_t;
+  using v4 = typename sycl::vec<T, 4>::vector_t;
+  using v8 = typename sycl::vec<T, 8>::vector_t;
```

Comment:
I am a little curious to hear @gmlueck thoughts here. Looks like `vector_t` becomes `std::array<T, N>` instead of a vector type. The specification calls `vector_t` an "underlying backend-native vector type defined by vector_t" while it is some opaque type. Is it okay for us to just use array as the `vector_t` type?